c10/core/TensorImpl.cpp

```diff
@@ -795,6 +795,15 @@ void TensorImpl::ShareExternalPointer(
   }
 }
 
+void TensorImpl::set_sym_sizes_and_strides(
+    c10::SymIntArrayRef sizes,
+    c10::SymIntArrayRef strides) {
+  has_symbolic_sizes_strides_ = true;
+  sizes_strides_policy_ = static_cast<uint8_t>(SizesStridesPolicy::CustomSizes);
```

Comment:
Another possibility is we could just assert that the policy is sufficiently high for symbolic sizes. This is OK because we will never store symbolic sizes in a stock tensor; we wouldn't be able to represent the data in question. It would have to be some sort of python sub class or some special subclass type like lazy tensor. Assuming this code works however, I wouldn't bother changing it for now.